gobblin-core-base/src/main/java/gobblin/writer/AsyncDataWriter.java

```diff
@@ -10,25 +10,33 @@
  * CONDITIONS OF ANY KIND, either express or implied.
  */
 
-package gobblin.kafka.writer;
+package gobblin.writer;
```

Comment:
I'd consider putting this in a separate package, say, gobblin.writer.async and adding a package-info.java class with the high-level picture description.